### DISCUSSION TITLE
Help me with start Keycloakify please

### DISCUSSION THREAD

**Discussion by Cherrywoood** [keycloakify/keycloakify#371] [Q&A]
Hello, I want to make custom themes and I found your solution. I installed the starter project, but to be honest, I have so many questions, I don't understand anything.
Here are some questions:
1. Can I use Keycloakify if I don't know TypeSctipt and I have JS project?
2. Can I use npm instead of yarn?
3. I don't really understand why the "yarn storybook" dependency is needed and what it does.
4. I don't really understand, do I need to use the starter project to restore your own pages for Keycloak? Do I need to make a separate project with only custom pages and inject a keycloakify dependency? Or I can inject a keycloakify dependency in my application, where there will be not only custom pages, but also any client-side application logic. How to do it right?
5. Do I need to use a Dockerfile to inject my themes into keycloak? Is it possible to use docker-compose and use volumes or environment variables?
6. I didn't really understand how to change CSS styles.

Please forgive me for stupid questions!

**Comment by garronej** [comment]
> Is it possible to use Keycloakify if I'm unfamiliar with TypeScript and am working on a JS project?

Yes, using TypeScript is entirely optional; it essentially serves as a scaffolding. Nonetheless, incorporating it can offer significant benefits for both you and your codebase.

> Can I use npm instead of yarn?

Indeed, you can use npm instead of yarn. However, many argue that yarn is superior due to its performance and features.

> I don't fully comprehend the purpose of the "yarn storybook" dependency and what it does.

The "yarn storybook" dependency aids in the isolated development of your login page. You can mock specific page configurations and ensure that your page is rendering correctly. Without it, verifying that everything is functioning as expected can be challenging. Storybook generates a web application similar to [this one](https://storybook.keycloakify.dev/) for your application. You have the option to discard storybook if you prefer.

> I'm unclear on whether I need to use the starter project to customize my Keycloak pages. Do I need to create a separate project solely for custom pages and inject a keycloakify dependency? Alternatively, can I inject a keycloakify dependency into my application, which will include not only custom pages but also client-side application logic? What's the correct approach?

The answer depends on your situation. If you already have a React application created with Create React App, it might be beneficial to set up Keycloakify within your existing project, thereby co-locating your Keycloak theme and app code.

If you are beginning a project from scratch and don't yet have a client application, you can use the starter project by default, which is configured to co-locate with an app.

However, if your app isn't built with React or if it's set up with Vite or Next, you may want to create a separate project for your Keycloak theme. In this case, start from the starter project and follow [this section](https://github.com/keycloakify/keycloakify-starter#standalone-keycloak-theme) to make it a standalone Keycloak theme.

> Is it necessary to use a Dockerfile to inject my themes into Keycloak? Can I use docker-compose and utilize volumes or environment variables?

Keycloakify generates a `keycloak-theme.jar` file designed to be loaded into Keycloak. It should be placed in `/opt/keycloak/providers`. 

The subsequent steps are up to you, but using Helm is a suggested method (information is displayed in the console when you build the theme). @willwill96 has also made [a video demonstrating how to create a Docker image](https://youtu.be/x3ux2JM1Bxk).  

Yes you can inject docker environement variable in your theme. [See doc](https://docs.keycloakify.dev/build-options#extrathemeproperties).  

> I'm finding it difficult to understand how to modify CSS styles.

You are free to use any tool you prefer to adjust the styles. If you are unsure, I recommend checking out [Material-UI](https://mui.com/); it offers numerous pre-configured UI components and the associated logic.

If you want a CSS-only solution, you can consider:

- Vanilla CSS or SCSS: Write `.css` files and import them into your code like `import "./myStyle.css"`.
- TSS: [TSS-React](https://www.tss-react.dev/) is a CSS-in-JS engine, and as its creator, I highly recommend it.
- [Tailwind UI](https://tailwindcss.com/)

Really, the method of modifying the CSS is beyond the scope of Keycloakify. You can approach it in the same way you would in any other React application.

I hope this information is helpful.

**Reply by Cherrywoood** [reply]
Thank you very much for the quick response!
So I understand correctly that the storybook is used during development, and in prod I can remove it?

May I don't close this discussion, while I set up keycloakify in my project, and ask questions as they come?

**Reply by garronej** [reply]
Yes Storybook is soely for dev purpose.

Sure you can keep this thread open

**Reply by Cherrywoood** [reply]
Hello. I want to implement keycloakify in my React app. I have read the documentation https://docs.keycloakify.dev/keycloakify-in-my-app and added the keycloakify dependency in my project.
I understand that I have to create the keycloak-theme directory myself and define all the pages that I need? Or do I need this directory from the start repository?

**Reply by garronej** [reply]
If I where you I'd copy the `keycloak-theme` dir from the the starter and put it in your project. Make sure everything build, then strart hacking frome here.  

But before anything, is you project a Create-react-app?

**Reply by Cherrywoood** [reply]
Are all the files needed in the keycloak-theme directory or can something be deleted? I just wanted to rewrite it somehow in JS. As I understand it, then it was necessary to make a page component, but simply compare it with the .ftl analogue? Or is it still better to copy, even if it uses TypeScript, to make it work?

> But before anything, is you project a Create-react-app?

I'm not sure. Are there others? I just created a project in WebStorm, but probably yes, it is.

**Reply by garronej** [reply]
Show me the `scripts` of your package.json, I'll tell you

**Reply by garronej** [reply]
> I just wanted to rewrite it somehow in JS.  

You mean, rewrite it from TS to JS?

**Reply by Cherrywoood** [reply]
> Show me the scripts of your package.json, I'll tell you

"scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "prepare": "copy-keycloak-resources-to-public",
    "build-keycloak-theme": "npm run build && keycloakify"
  },

> You mean, rewrite it from TS to JS?

Yes

**Reply by garronej** [reply]
Ok, your are indeed using an CRA.  

> Yes  

I would highly recommend reconsidering if you're planning to use vanilla JavaScript. Although TypeScript may appear daunting at first, opting for vanilla JS can be likened to starting a game on hardcore mode, and this could potentially be a choice you regret in the future.

**Reply by Cherrywoood** [reply]
Ok, thank you!

**Reply by garronej** [reply]
If you make you project open source I can tell you if you're going in the right direction if you want

**Reply by Cherrywoood** [reply]
Thanks a lot!
Here is my repository: https://github.com/Cherrywoood/react-keycloak
I added keycloakify and srotybook (npm i --save keyloakify and npm i --save storybook), but I get the error "Cannot find module '@storybook/react' or its corresponding type declarations."

**Reply by Cherrywoood** [reply]
Hello, there are a few more questions that I found while trying to work with keycloakify:
1. Where should I define styles? As I see it, there is login.css in public, but there are also styles in KcApp.css? Where can I define them?
2. If I want to customize only the login page, which files should I keep and which can I remove from keycloak-theme?
3. Do I understand correctly that MyExtraPage is some kind of additional pages that are not pages from Keycloak?
4. Can I remove i18n if I don't need it?
5. Is it possible to change the html content of login.tsx or other pages?
6. I don't really understand why download default themes with npx download-builtin-keycloak-theme

**Reply by Saadkhh** [reply]
Hi @garronej

when using pure css to add styles to my login page, i get the following error
![image](https://github.com/keycloakify/keycloakify/assets/107172255/d16e7f08-f39d-49d0-8459-5cdc3b1fe73e)

after i place the .jar in my /opt/keycloak/providers and using the keycloakify-starter-theme

i have tried using inline styles and importing css files just as you described in your answer 

> Vanilla CSS or SCSS: Write .css files and import them into your code like import "./myStyle.css".

ive been stuck on this for hours, do you have an idea on how to solve this?

**Reply by garronej** [reply]
If you share your repo, or a minimal repo. I would probably be able to tell you what's wrong.

**Reply by Saadkhh** [reply]
Hey @garronej, when building the theme in vscode, i always get the "Refused to apply style ... MIME" error. When using Github Codespaces to build the theme, no error occurs. 

But i stumbled upon another problem: When using "keyclokify-starter" theme for the account management console, the user cannot change its email, firstname and lastname. You simply cannot insert new text or change the current firstname, lastname or email.
But when using "keycloak" theme, everthing works fine. 

I didnt even change anything in the code. I followed the keycloakify-starter docu and used these [commands](https://github.com/keycloakify/keycloakify-starter#standalone-keycloak-theme), build the jar and used it in my keycloak container. 

Do you have any idea on how  to fix this?

**Reply by garronej** [reply]
Hi @Saadkhh,  
What version of keycloakify and Keycloak are you using?

**Reply by Saadkhh** [reply]
using keycloakify-starter 4.7.2 and keycloak v19.0.3

**Reply by garronej** [reply]
@Saadkhh,  

I can't reproduce.  
When I take the latest version of the starter and I follow the instructions printed on the console after I build the theme, it works:  

<img width="1538" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/94a60480-c00e-40db-9150-cd541e185e90">

**Reply by garronej** [reply]
As you can see, the field are disabled when editUserNameAllowed is false. This is somethis that you edit in you realm setting:  

<img width="739" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/bf06747a-346a-4495-b73e-62edb81ca8fe">

**Comment by Saadkhh** [comment]
i dont use  ./build_keycloak/start_keycloak_testing_container.sh, i have my own container in which i copy the .jar inside. I have tried different keycloak versions but i still cannot change the firstname, lastname, email. Is my generated .jar faulty?

**Reply by garronej** [reply]
No, I doubt it.  
Wether you test with the jar or with the commend it shouldn't affect the result.  
If when you select `keycloak` as theme (not `keycloak.v2`) it works, it should mean that your realm configuration is fine.  
Then it boils down to your react code.  

Can you show me you `kcContext`?  

<img width="1727" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/536de7c4-e1b3-4f7a-84a7-01b7afc42463">

**Reply by garronej** [reply]
there's maybe something related to the second point of the v8 migration guide https://docs.keycloakify.dev/migration-guides/v7-greater-than-v8

**Reply by Saadkhh** [reply]
heres my kcContext

```json
{
    "features": {
        "log": false,
        "identityFederation": false,
        "authorization": true,
        "passwordUpdateSupported": true
    },
    "messagesPerField": {},
    "stateChecker": "JlJYJOCuBLbF0_G2pu5cttppM7vJ12AloZVXA8z_-6I",
    "realm": {
        "supportedLocales": [],
        "displayName": "myrealm",
        "userManagedAccessAllowed": true,
        "displayNameHtml": "myrealm",
        "editUsernameAllowed": false,
        "internationalizationEnabled": false,
        "name": "myrealm",
        "registrationEmailAsUsername": false
    },
    "properties": {
        "parent": "keycloak",
        "kcButtonLargeClass": "btn-lg",
        "locales": "ar,ca,cs,da,de,en,es,fr,fi,hu,it,ja,lt,nl,no,pl,pt-BR,ru,sk,sv,tr,zh-CN",
        "import": "common/keycloak",
        "kcButtonPrimaryClass": "btn-primary",
        "stylesCommon": "node_modules/patternfly/dist/css/patternfly.min.css node_modules/patternfly/dist/css/patternfly-additions.min.css",
        "foo": "bar",
        "styles": "css/account.css",
        "kcButtonClass": "btn",
        "kcButtonDefaultClass": "btn-default"
    },
    "url": {
        "resourcesPath": "/resources/3dlhc/account/keycloakify-starter",
        "resourceUrl": "http://localhost:8080/realms/myrealm/account/resource",
        "resourcesCommonPath": "/resources/3dlhc/account/keycloakify-starter/resources_common",
        "logUrl": "http://localhost:8080/realms/myrealm/account/log",
        "socialUrl": "http://localhost:8080/realms/myrealm/account/identity",
        "accountUrl": "http://localhost:8080/realms/myrealm/account/",
        "sessionsUrl": "http://localhost:8080/realms/myrealm/account/sessions",
        "totpUrl": "http://localhost:8080/realms/myrealm/account/totp",
        "logoutUrl": "http://localhost:8080/realms/myrealm/protocol/openid-connect/logout?post_logout_redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Frealms%2Fmyrealm%2Faccount%2F&id_token_hint=eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJkaTlJRnB0ZW5RUTUwd3YtV2lHR2k1MW1pcElWZ2t5ZWt6Q05sZ1l6aHhZIn0.eyJleHAiOjE2OTY4MjA1NTAsImlhdCI6MTY5Njc4NDU5MCwiYXV0aF90aW1lIjowLCJqdGkiOiJkY2YyM2JiOS1hNjRjLTQ1Y2YtODQ1My05ZmQxZTlhODBkMmMiLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvcmVhbG1zL215cmVhbG0iLCJhdWQiOiJhY2NvdW50Iiwic3ViIjoiNGUwZmJhYjgtYzJhMi00YmE5LThmZmQtYjZhMzdmMzBhMDQxIiwidHlwIjoiSUQiLCJhenAiOiJhY2NvdW50Iiwic2Vzc2lvbl9zdGF0ZSI6IjliZWNjZGM4LWNiYmItNDE0OS05ZTEyLWM2ODVlZTM1NGI1NyIsImFjciI6IjEiLCJzaWQiOiI5YmVjY2RjOC1jYmJiLTQxNDktOWUxMi1jNjg1ZWUzNTRiNTciLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJTYWFkIHNhYWQiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJzYWFkIiwiZ2l2ZW5fbmFtZSI6IlNhYWQiLCJmYW1pbHlfbmFtZSI6InNhYWQiLCJlbWFpbCI6InNhYWRraEB3ZWIuZGUifQ.kTff5KeRkfNxThhAUBNVZ3CH8vv2rA8thsfJf20pu5VfNddJcGbk1hRDzJL4xKO6GxzQzqx3RY4gperQk9HqxxlasBUnxDPHqdkF6wj1-Gd3FAgebQLSjiIdsW2K5nep8fOWOAEn9-OSvMp8GqDcnuRaKiOkY-93iVaS5ism-au_VYOdv-EGOwMZxCsivbbxJwujnE8k26CeRfKIVTMoRKw-7NVrdWByZTe_eW-l_fR50I9jOcL4vlGoVPaWCFaax-biaXZ-h7ZKA8vS9ISgMPLNNc6mJOdx3zk4tpgmjHxgW3PALvMrkcyIajs0ySdiBNoAYutrur9TKrmnNK55dQ",
        "applicationsUrl": "http://localhost:8080/realms/myrealm/account/applications",
        "passwordUrl": "http://localhost:8080/realms/myrealm/account/password"
    },
    "account": {
        "lastName": "saad",
        "firstName": "Saad",
        "attributes": {
            "firstName": "Saad",
            "lastName": "saad",
            "email": "saad@web.de",
            "username": "saad"
        },
        "email": "saad@web.de",
        "username": "saad"
    },
    "keycloakifyVersion": "8.1.2",
    "themeVersion": "4.7.2",
    "themeType": "account",
    "themeName": "keycloakify-starter",
    "pageId": "account.ftl"
}
```

**Reply by garronej** [reply]
Well I can see in your relam configuration that:  

<img width="342" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/d9b60284-a45c-4d8e-b5c2-b3a9e6876b4f">

**Reply by Saadkhh** [reply]
yes, even with edit username enabled i cannot change the text inside the textfield: 
![image](https://github.com/keycloakify/keycloakify/assets/107172255/fbbc0c31-3672-492e-8df5-916d5faab0a8)

with "keycloak" theme: 
![image](https://github.com/keycloakify/keycloakify/assets/107172255/561d15a0-e640-426f-92ac-9aa1705348f1)

i really dont know what im doing wrong here...

**Reply by garronej** [reply]
What do you mean exactly by "i cannot change the text inside the textfield"?  
Have you tried running your page locally with `mockPageId: "account.ftl` ?  
Do you have the problem in that case?

**Comment by garronej** [comment]
@Saadkhh Oh! Sorry! This is my bad, I used value instead of defaultValue, let me release a patch! Thanks for noticing!

**Reply by Saadkhh** [reply]
Hi @garronej, thanks, it works now with 4.7.3! 

But i have stumbled upon another problem:
When using "keycloak-starter" theme for account console the logout doesnt seem to work. I always get this after pressing "Sign Out": 
![image](https://github.com/keycloakify/keycloakify/assets/107172255/6d180c55-9171-4626-9428-3220000a8211)
and this is how the URL looks like after pressing "Sign Out":  http://localhost:8080/realms/myrealm/protocol/openid-connect/logout?post_logout_redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Frealms%2Fmyrealm%2Faccount%2F&amp;id_token_hint={JWT_TOKEN}

(i replaced the actual jwt token)

When using "keycloak" theme for account console, the logout works and redirects me to the login page.

Do you have any idea on how to solve this?

**Comment by garronej** [comment]
@Cherrywoood It should be fixed int the latest release.  
Maybe your problem releated to MIM type can be solved by running `rm -rf public/keycloak-resources && yarn build-keycloak-theme`  

Sorry about the bad dev experience :/

**Reply by com-admin** [reply]
facing MIME issue in the 8.1.2 keycloakify and 4.7.2 keycloakify-starter with 22.0.3 keycloak docker. The resource from “http://localhost:8080/resources/0vyio/login/keycloakify-starter/resources_common/node_modules/patternfly/dist/css/patternfly-additions.min.css” was blocked due to MIME type (“”) mismatch (X-Content-Type-Options: nosniff) in both firefox and chrome
This error does not happen if I use the default keycloak theme, only occurs when I set the 'keycloak-starter' theme. The file is actually present in this location http://localhost:8080/resources/0vyio/common/keycloak/node_modules/patternfly/dist/css/patternfly-additions.min.css

**Reply by garronej** [reply]
Please try Keycloakify 8.1.3, please reamove public/keycloak-resources and build_keycloak and rebuild you theme again with `yarn build-keycloak-theme`

**Reply by com-admin** [reply]
hi, tried with 8.1.3, now the patternfly MIME issue is not happening, but now getting `The resource from “http://localhost:8080/resources/ih4rn/login/keycloakify-starter/resources_common/lib/zocial/zocial.css” was blocked due to MIME type (“”) mismatch (X-Content-Type-Options: nosniff).`
my kccontext:
```
{
  "execution": "ef624fd2-729c-4f73-bb0c-1e1fdf722e6c",
  "social": {
    "displayInfo": false
  },
  "auth": {
    "showResetCredentials": false,
    "authenticationSelections": [
      {
        "helpText": "auth-username-password-form-help-text",
        "displayName": "auth-username-password-form-display-name",
        "iconCssClass": "kcAuthenticatorDefaultClass",
        "authExecId": "ef624fd2-729c-4f73-bb0c-1e1fdf722e6c",
        "authenticationExecution": {
          "authenticatorFlow": false,
          "conditional": false,
          "required": true,
          "enabled": true,
          "disabled": false,
          "id": "ef624fd2-729c-4f73-bb0c-1e1fdf722e6c",
          "parentFlow": "f65547dd-4ad0-4329-b270-7f41e827466e",
          "authenticator": "auth-username-password-form",
          "alternative": false,
          "requirement": {
            "stableIndex": "0"
          },
          "priority": "10"
        }
      }
    ],
    "showUsername": false,
    "showTryAnotherWayLink": false
  },
  "login": {},
  "url": {
    "oauthAction": "/realms/myrealm/login-actions/authenticate",
    "resourcesPath": "/resources/ih4rn/login/keycloakify-starter",
    "loginAction": "http://localhost:8080/realms/myrealm/login-actions/authenticate?session_code=2hjkAMdVQepoFE8H9G2bQpaqhbENpQ-cbTExCvjSIaI&execution=ef624fd2-729c-4f73-bb0c-1e1fdf722e6c&client_id=myclient&tab_id=jo35bUhLjLE",
    "loginUrl": "/realms/myrealm/login-actions/authenticate?client_id=myclient&tab_id=jo35bUhLjLE",
    "resourcesCommonPath": "/resources/ih4rn/login/keycloakify-starter/resources_common",
    "firstBrokerLoginUrl": "/realms/myrealm/login-actions/first-broker-login?client_id=myclient&tab_id=jo35bUhLjLE",
    "registrationUrl": "/realms/myrealm/login-actions/registration?client_id=myclient&tab_id=jo35bUhLjLE",
    "logoutConfirmAction": "/realms/myrealm/protocol/openid-connect/logout/logout-confirm?client_id=myclient&tab_id=jo35bUhLjLE",
    "registrationAction": "http://localhost:8080/realms/myrealm/login-actions/authenticate?session_code=2hjkAMdVQepoFE8H9G2bQpaqhbENpQ-cbTExCvjSIaI&execution=ef624fd2-729c-4f73-bb0c-1e1fdf722e6c&client_id=myclient&tab_id=jo35bUhLjLE",
    "loginResetCredentialsUrl": "/realms/myrealm/login-actions/reset-credentials?client_id=myclient&tab_id=jo35bUhLjLE",
    "resourcesUrl": "/resources/ih4rn?client_id=myclient&tab_id=jo35bUhLjLE/login/keycloakify-starter",
    "loginRestartFlowUrl": "/realms/myrealm/login-actions/restart?client_id=myclient&tab_id=jo35bUhLjLE",
    "oauth2DeviceVerificationAction": "/realms/myrealm/login-actions/authenticate"
  },
  "messagesPerField": {},
  "client": {
    "clientId": "myclient",
    "description": "",
    "baseUrl": "",
    "name": "",
    "attributes": {
      "oidc.ciba.grant.enabled": "false",
      "backchannel.logout.session.required": "true",
      "login_theme": "keycloakify-starter",
      "post.logout.redirect.uris": "https://www.keycloak.org/app*##http://localhost*",
      "oauth2.device.authorization.grant.enabled": "false",
      "display.on.consent.screen": "false",
      "backchannel.logout.revoke.offline.tokens": "false"
    }
  },
  "realm": {
    "displayName": "myrealm",
    "registrationAllowed": false,
    "displayNameHtml": "myrealm",
    "resetPasswordAllowed": false,
    "resetCredentialsActionTokenLifespanMinutes": "5",
    "password": true,
    "actionTokenGeneratedByUserLifespanMinutes": "5",
    "internationalizationEnabled": false,
    "idpVerifyAccountLinkActionTokenLifespanMinutes": "5",
    "verifyEmailActionTokenLifespanMinutes": "5",
    "loginWithEmailAllowed": true,
    "duplicateEmailsAllowed": false,
    "editUsernameAllowed": false,
    "name": "myrealm",
    "rememberMe": false,
    "registrationEmailAsUsername": false,
    "identityFederationEnabled": false
  },
  "scripts": [],
  "properties": {
    "kcLogoIdP-facebook": "fa fa-facebook",
    "parent": "keycloak",
    "kcAuthenticatorOTPClass": "fa fa-mobile list-view-pf-icon-lg",
    "kcLogoIdP-bitbucket": "fa fa-bitbucket",
    "kcAuthenticatorWebAuthnClass": "fa fa-key list-view-pf-icon-lg",
    "kcWebAuthnDefaultIcon": "pficon pficon-key",
    "kcLogoIdP-stackoverflow": "fa fa-stack-overflow",
    "kcSelectAuthListItemClass": "pf-l-stack__item select-auth-box-parent pf-l-split",
    "kcLogoIdP-microsoft": "fa fa-windows",
    "kcLocaleItemClass": "pf-c-dropdown__menu-item",
    "kcLoginOTPListItemHeaderClass": "pf-c-tile__header",
    "kcLoginOTPListItemIconBodyClass": "pf-c-tile__icon",
    "kcInputHelperTextAfterClass": "pf-c-form__helper-text pf-c-form__helper-text-after",
    "kcFormClass": "form-horizontal",
    "kcSelectAuthListClass": "pf-l-stack select-auth-container",
    "kcInputClassRadioCheckboxLabelDisabled": "pf-m-disabled",
    "kcSelectAuthListItemIconClass": "pf-l-split__item select-auth-box-icon",
    "kcRecoveryCodesWarning": "kc-recovery-codes-warning",
    "kcFormSettingClass": "login-pf-settings",
    "kcWebAuthnBLE": "fa fa-bluetooth-b",
    "kcInputWrapperClass": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
    "kcSelectAuthListItemArrowIconClass": "fa fa-angle-right fa-lg",
    "meta": "viewport==width=device-width,initial-scale=1",
    "styles": "css/login.css",
    "kcFeedbackAreaClass": "col-md-12",
    "kcLogoIdP-google": "fa fa-google",
    "kcCheckLabelClass": "pf-c-check__label",
    "kcSelectAuthListItemFillClass": "pf-l-split__item pf-m-fill",
    "kcAuthenticatorDefaultClass": "fa fa-list list-view-pf-icon-lg",
    "kcLogoIdP-gitlab": "fa fa-gitlab",
    "kcFormAreaClass": "col-sm-10 col-sm-offset-1 col-md-8 col-md-offset-2 col-lg-8 col-lg-offset-2",
    "kcFormButtonsClass": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
    "kcInputClassRadioLabel": "pf-c-radio__label",
    "kcAuthenticatorWebAuthnPasswordlessClass": "fa fa-key list-view-pf-icon-lg",
    "kcSelectAuthListItemHeadingClass": "pf-l-stack__item select-auth-box-headline pf-c-title",
    "kcInfoAreaClass": "col-xs-12 col-sm-4 col-md-4 col-lg-5 details",
    "kcLogoLink": "http://www.keycloak.org",
    "kcContainerClass": "container-fluid",
    "kcSelectAuthListItemTitle": "select-auth-box-paragraph",
    "kcHtmlClass": "login-pf",
    "kcLoginOTPListItemTitleClass": "pf-c-tile__title",
    "locales": "ar,ca,cs,da,de,el,en,es,fa,fr,fi,hu,it,ja,lt,nl,no,pl,pt-BR,ru,sk,sv,tr,zh-CN",
    "kcLogoIdP-openshift-v4": "pf-icon pf-icon-openshift",
    "kcWebAuthnUnknownIcon": "pficon pficon-key unknown-transport-class",
    "kcFormSocialAccountNameClass": "kc-social-provider-name",
    "kcLogoIdP-openshift-v3": "pf-icon pf-icon-openshift",
    "kcLoginOTPListInputClass": "pf-c-tile__input",
    "kcWebAuthnUSB": "fa fa-usb",
    "kcInputClassRadio": "pf-c-radio",
    "kcWebAuthnKeyIcon": "pficon pficon-key",
    "kcFeedbackInfoIcon": "fa fa-fw fa-info-circle",
    "kcCommonLogoIdP": "kc-social-provider-logo kc-social-gray",
    "stylesCommon": "web_modules/@patternfly/react-core/dist/styles/base.css web_modules/@patternfly/react-core/dist/styles/app.css node_modules/patternfly/dist/css/patternfly.min.css node_modules/patternfly/dist/css/patternfly-additions.min.css lib/pficon/pficon.css",
    "kcRecoveryCodesActions": "kc-recovery-codes-actions",
    "kcFormGroupHeader": "pf-c-form__group",
    "kcFormSocialAccountSectionClass": "kc-social-section kc-social-gray",
    "kcLogoIdP-instagram": "fa fa-instagram",
    "kcAlertClass": "pf-c-alert pf-m-inline",
    "kcHeaderClass": "login-pf-page-header",
    "kcLabelWrapperClass": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
    "kcFormSocialAccountLinkClass": "pf-c-login__main-footer-links-item-link",
    "kcLocaleMainClass": "pf-c-dropdown",
    "kcTextareaClass": "form-control",
    "kcButtonBlockClass": "pf-m-block",
    "kcButtonClass": "pf-c-button",
    "kcWebAuthnNFC": "fa fa-wifi",
    "kcLocaleClass": "col-xs-12 col-sm-1",
    "kcInputClassCheckboxInput": "pf-c-check__input",
    "kcFeedbackErrorIcon": "fa fa-fw fa-exclamation-circle",
    "kcInputLargeClass": "input-lg",
    "kcInputErrorMessageClass": "pf-c-form__helper-text pf-m-error required kc-feedback-text",
    "kcRecoveryCodesList": "kc-recovery-codes-list",
    "kcFormSocialAccountListClass": "pf-c-login__main-footer-links kc-social-links",
    "kcAlertTitleClass": "pf-c-alert__title kc-feedback-text",
    "kcAuthenticatorPasswordClass": "fa fa-unlock list-view-pf-icon-lg",
    "kcCheckInputClass": "pf-c-check__input",
    "kcLogoIdP-linkedin": "fa fa-linkedin",
    "kcLogoIdP-twitter": "fa fa-twitter",
    "kcFeedbackWarningIcon": "fa fa-fw fa-exclamation-triangle",
    "kcResetFlowIcon": "pficon pficon-arrow fa",
    "kcSelectAuthListItemIconPropertyClass": "fa-2x select-auth-box-icon-properties",
    "kcFeedbackSuccessIcon": "fa fa-fw fa-check-circle",
    "kcLoginOTPListClass": "pf-c-tile",
    "kcSrOnlyClass": "sr-only",
    "kcFormSocialAccountListGridClass": "pf-l-grid kc-social-grid",
    "kcButtonDefaultClass": "btn-default",
    "kcFormGroupErrorClass": "has-error",
    "kcSelectAuthListItemDescriptionClass": "pf-l-stack__item select-auth-box-desc",
    "kcSelectAuthListItemBodyClass": "pf-l-split__item pf-l-stack",
    "import": "common/keycloak",
    "kcWebAuthnInternal": "pficon pficon-key",
    "kcSelectAuthListItemArrowClass": "pf-l-split__item select-auth-box-arrow",
    "kcCheckClass": "pf-c-check",
    "foo": "bar",
    "kcContentClass": "col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3 col-lg-6 col-lg-offset-3",
    "kcLogoClass": "login-pf-brand",
    "kcLoginOTPListItemIconClass": "fa fa-mobile",
    "kcLoginClass": "login-pf-page",
    "kcSignUpClass": "login-pf-signup",
    "kcButtonLargeClass": "btn-lg",
    "kcFormCardClass": "card-pf",
    "kcLocaleListClass": "pf-c-dropdown__menu pf-m-align-right",
    "kcInputClass": "pf-c-form-control",
    "kcFormGroupClass": "form-group",
    "kcLogoIdP-paypal": "fa fa-paypal",
    "kcInputClassCheckbox": "pf-c-check",
    "kcRecoveryCodesConfirmation": "kc-recovery-codes-confirmation",
    "kcInputClassRadioInput": "pf-c-radio__input",
    "kcFormSocialAccountListButtonClass": "pf-c-button pf-m-control pf-m-block kc-social-item kc-social-gray",
    "kcInputClassCheckboxLabel": "pf-c-check__label",
    "kcFormOptionsClass": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
    "kcFormHeaderClass": "login-pf-header",
    "kcFormSocialAccountGridItem": "pf-l-grid__item",
    "kcButtonPrimaryClass": "pf-m-primary",
    "kcInputHelperTextBeforeClass": "pf-c-form__helper-text pf-c-form__helper-text-before",
    "kcLogoIdP-github": "fa fa-github",
    "kcLabelClass": "pf-c-form__label pf-c-form__label-text"
  },
  "keycloakifyVersion": "8.1.3",
  "themeVersion": "4.7.2",
  "themeType": "login",
  "themeName": "keycloakify-starter",
  "pageId": "login.ftl"
}
```

**Reply by garronej** [reply]
Can you tell me if the Strarter App that is deployed here works for you? https://starter.keycloakify.dev/

**Reply by com-admin** [reply]
yes, I'm able to login to  https://starter.keycloakify.dev/  successfully

**Reply by com-admin** [reply]
commenting out these lines make the login page load fine
```
  const { isReady } = usePrepareTemplate({
        "doFetchDefaultThemeResources": doUseDefaultCss,
        "styles": [
            // `${url.resourcesCommonPath}/node_modules/patternfly/dist/css/patternfly.min.css`,
            // `${url.resourcesCommonPath}/node_modules/patternfly/dist/css/patternfly-additions.min.css`,
            // `${url.resourcesCommonPath}/lib/zocial/zocial.css`,
            `${url.resourcesPath}/css/login.css`
        ],
        "htmlClassName": getClassName("kcHtmlClass"),
        "bodyClassName": getClassName("kcBodyClass")
    });
```

**Reply by com-admin** [reply]
in my kccontext there is `"resourcesCommonPath": "/resources/ih4rn/login/keycloakify-starter/resources_common",`
but this `resources_common` folder is missing in my `build_keycloak\src\main\resources\theme\keycloakify-starter\login`

**Reply by com-admin** [reply]
updating the resourcesCommon paths to resourcesPath and manually copying those css files into `build_keycloak\src\main\resources\theme\keycloakify-starter\login\resources\css` works

```
    const { isReady } = usePrepareTemplate({
        "doFetchDefaultThemeResources": doUseDefaultCss,
        "styles": [
            `${url.resourcesPath}/css/patternfly.min.css`,
            `${url.resourcesPath}/css/patternfly-additions.min.css`,
            `${url.resourcesPath}/css/zocial.css`,
            `${url.resourcesPath}/css/login.css`
        ],
        "htmlClassName": getClassName("kcHtmlClass"),
        "bodyClassName": getClassName("kcBodyClass")
    });
```

**Reply by garronej** [reply]
> yes, I'm able to login to https://starter.keycloakify.dev/ successfully  

Well this is a deployment of the last version of the starter, as is, so there shouldn't be any needs to change the code but there is maybe a bug happening with your specific dev environement.  
Can you try to clone from scratch the starter project, without changing anything and tell me if the problem do hapen (please start from git clone https://github.com/keycloakify/keycloakify-starter to exclude any cache releated issues).  

If it does can you tell me what version of node and what OS you are using?
